sim/hunter/items.go

```diff
@@ -569,6 +570,118 @@ func init() {
+		}
+
+		if !hasMeleeSpecialist {
+			rangedTrigger := core.MakeProcTriggerAura(&hunter.Unit, core.ProcTrigger{
+				Name:       "Strand Of Fate - Ranged Trigger",
+				Callback:   core.CallbackOnSpellHitDealt,
+				Outcome:    core.OutcomeLanded,
+				ProcMask:   core.ProcMaskRanged,
+				ProcChance: 0.1,
+				Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if !icd.IsReady(sim) {
+						return
+					}
+					stacksAura.Activate(sim)
+					stacksAura.AddStack(sim)
+				},
+			})
+
+			hunter.ItemSwap.RegisterProc(StringsOfFate, rangedTrigger)
+		} else {
+			stacksAura.OnStacksChange = func(aura *core.Aura, sim *core.Simulation, oldStacks, newStacks int32) {
+				strengthChange := float64(20 * (newStacks - oldStacks))
+				hunter.Unit.AddStatDynamic(sim, stats.Strength, strengthChange)
+			}
+
+			meleeTrigger := core.MakeProcTriggerAura(&hunter.Unit, core.ProcTrigger{
+				Name:       "Strand Of Fate - Melee Trigger",
+				Callback:   core.CallbackOnSpellHitDealt,
+				Outcome:    core.OutcomeLanded,
+				ProcMask:   core.ProcMaskMelee,
+				ProcChance: 0.1,
+				Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+					if !icd.IsReady(sim) {
+						return
+					}
+					stacksAura.Activate(sim)
+					stacksAura.AddStack(sim)
+				},
+			})
+
+			hunter.ItemSwap.RegisterProc(StringsOfFate, meleeTrigger)
+		}
+
+		// When using the melee version and activating the bow your next strike applies a special Serpent Sting effect.
+		var strands int32
+		serpentStrikeAura := hunter.RegisterAura(core.Aura{
+			Label:    "Strand Of Fate - Serpent Sting on Next Strike",
+			ActionID: core.ActionID{SpellID: 1232976},
+			Duration: time.Second * 20,
+			OnGain: func(aura *core.Aura, sim *core.Simulation) {
+				strands = stacksAura.GetStacks()
+			},
+			OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if spell.Matches(ClassSpellMask_HunterStrikes) && result.Landed() {
+					hunter.SoFSerpentSting[strands].Cast(sim, result.Target)
+					aura.Deactivate(sim)
+				}
+			},
+		})
+
+		// Gain 40 strength per stack of Strand of Fate consumed
+		strengthAura := hunter.NewTemporaryStatsAura("Strand Of Fate - Strength", core.ActionID{SpellID: 1232969}, stats.Stats{stats.Strength: float64(40 * stacksAura.GetStacks())}, time.Second * 20)
+
+		// The bow active
+		spell := hunter.RegisterSpell(core.SpellConfig{
+			ActionID: core.ActionID{SpellID: 1231604},
+			Flags:    core.SpellFlagNoOnCastComplete | core.SpellFlagOffensiveEquipment,
+			Cast: core.CastConfig{
+				CD: core.Cooldown{
+					Timer:    hunter.NewTimer(),
+					Duration: time.Second * 80,
+				},
+			},
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				if hasMeleeSpecialist && stacksAura.IsActive() {
+					strengthAura.Activate(sim)
+					serpentStrikeAura.Activate(sim)
+					stacksAura.Deactivate(sim)
+					icd.Use(sim)
```

Comment:
Sorry I completely misread this. it looks like this should not refresh the strengthAura and when activating the `strengthAura` should set the duration to 21s times the number of stacks consumed

![image](https://github.com/user-attachments/assets/d7b25da8-519d-48f1-8ab8-a0b1dd36fd4a)

Reply:
I also think we probably shouldn't use the ICD in this active. Normally the ICD is just a part of the proc trigger and the active doesn't mention anything about it